tests/JobDefinitionFactoryTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests;
+
+use App\JobDefinitionFactory;
+use Keboola\DockerBundle\Docker\Component;
+use Keboola\JobQueueInternalClient\JobFactory;
+use Keboola\JobQueueInternalClient\JobFactory\Job;
+use Keboola\ObjectEncryptor\ObjectEncryptor;
+use Keboola\ObjectEncryptor\ObjectEncryptorFactory;
+use Keboola\StorageApi\Client;
+use PHPUnit\Framework\TestCase;
+
+class JobDefinitionFactoryTest extends TestCase
+{
+    public function testCreateJobDefinitionWithConfigData(): void
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+        ];
+
+        $jobDefinitions = $this->createJobDefinitions($jobData, $configData);
+
+        self::assertCount(1, $jobDefinitions);
+        $jobDefinition = $jobDefinitions[0];
+
+        self::assertSame($jobData['configId'], $jobDefinition->getConfigId());
+        self::assertSame($jobData['componentId'], $jobDefinition->getComponentId());
+        self::assertSame('bar', $jobDefinition->getConfiguration()['runtime']['foo'] ?? null);
```

Comment:
?? null je tam proč?

Reply:
Aby se ten assert vyhodnotil spravne i pokud to v configuration chybi. V aktualni podobe dostanes neco jako "`null` does not match expected `bar`". Pokud by tam `?? null` nebylo, vybihala by tam (navic) nejaka NOTICE, ze pristupujes k nedefinovanymu indexu na poli.

Prijde mi to elegantnejsi, nez explicitne psat 3 asserty, ze je tam klic `runtime`, pak ze je tam klic `foo` a pak konecne, ze `foo` ma hodnotu `bar`.

Reply:
no přijde mi to takový divný, ale asi nedokážu rozumně říct proč, takže je to oukej 😆